sanic/response.py

```diff
@@ -359,8 +360,11 @@ def redirect(to, headers=None, status=302,
     """
     headers = headers or {}
 
+    # URL Quote the URL before redirecting
+    safe_to = quote_plus(to, safe=":/#?&=@[]!$&'()*+,;")
```

Comment:
Hi @aveao and @r0fls , I have a question regarding the safe character. 

When creating the redirected URL, I have already escaped the query params (using `quote_plus` via `urllib.parse.urlencode()`. This line would escape `%` char for the 2nd time causing a problem to unquote the redirected URL. Would it be possible to mark `%` as a safe char?

Also, I think there are two `&`s in the `safe` string.